File: packages/agentmark/src/prompts/text.ts
import { Adapter, TemplateEngine, JSONObject, PromptMetadata, AdapterTextOutput, TextConfig } from "../types";
import { Prompt } from "./index";

export interface TextPromptInterface<Props extends JSONObject, Result extends string, A extends Adapter = Adapter> 
  extends Prompt<Props, Result, A> {
  format(props: Props, options?: JSONObject): Promise<AdapterTextOutput<A, Result>>;
}

export class TextPrompt<
  Props extends JSONObject,
  Result extends string,
  A extends Adapter = Adapter
> implements TextPromptInterface<Props, Result, A> {
  protected templateEngine: TemplateEngine;
  protected adapter: A;
  protected path: string | undefined;
  public template: unknown;
  
  constructor(template: unknown, templateEngine: TemplateEngine, adapter: A, path?: string | undefined) {
    this.template = template;
    this.templateEngine = templateEngine;
    this.adapter = adapter;
    this.path = path;
  }

  async format(
    props: Props, 
    options: JSONObject = {}
  ): Promise<AdapterTextOutput<A, Result>> {
    const compiledTemplate = await this.templateEngine.compile(this.template, props) as TextConfig;
    const metadata: PromptMetadata = { props, path: this.path, template: this.template };
    const result = this.adapter.adaptText<Result>(compiledTemplate, options, metadata);
    return result as AdapterTextOutput<A, Result>;
  }
} 

File: packages/prompt-core-python/src/agentmark/prompt_core/prompts/text.py
"""Text prompt class."""

import inspect
from typing import Any

from ..schemas import TextConfigSchema
from .base import BasePrompt


class TextPrompt(BasePrompt[TextConfigSchema]):
    """Text prompt for chat completions."""

    async def format(
        self,
        props: dict[str, Any] | None = None,
        **options: Any,
    ) -> Any:
        """Format the text prompt.

        Args:
            props: Props to pass to the template
            **options: Additional adapter options

        Returns:
            Adapted text prompt output
        """
        compiled = await self._compile(props)
        adapt_options = self._build_adapt_options(options)
        result = self._adapter.adapt_text(compiled, adapt_options, self._metadata(props))
        # Support both sync and async adapters
        if inspect.iscoroutine(result):
            result = await result
        return result
